### PR TITLE
Easy speed-ups for tests of units.

### PR DESCRIPTION
**Pull request by mhvk**
Trivial changes to remove the units module from having entries in the top 50 slowest.

### CODE REVIEW

**Comment by bsipocz** [comment]
Not critical, but this should ideally go back to the backport branch.

File: astropy/units/tests/test_units.py [hunk 1]
```diff
@@ -348,10 +348,9 @@ def test_compose_roundtrip(unit):
 @pytest.mark.parametrize('unit', sorted(COMPOSE_CGS_TO_SI, key=_unit_as_str),
                          ids=_unit_as_str)
 def test_compose_cgs_to_si(unit):
-    for iter in range(10):
-        si = unit.to_system(u.si)
-        assert [x.is_equivalent(unit) for x in si]
-        assert si[0] == unit.si
+    si = unit.to_system(u.si)
+    assert [x.is_equivalent(unit) for x in si]
+    assert si[0] == unit.si
```
Comment (astrofrog) [inline astropy/units/tests/test_units.py]: Wait what, why was this in a loop??

File: astropy/units/tests/test_units.py [hunk 2]
```diff
@@ -414,7 +413,10 @@ def test_compose_failed():
 
 
 def test_compose_fractional_powers():
-    x = (u.kg / u.s ** 3 * u.au ** 2.5 / u.yr ** 0.5 / u.sr ** 2)
+    # Warning: with a complicated unit, this test becomes very slow;
+    # e.g., x = (u.kg / u.s ** 3 * u.au ** 2.5 / u.yr ** 0.5 / u.sr ** 2)
+    # takes 3 s
```
Comment (astrofrog) [inline astropy/units/tests/test_units.py]: We should open an issue for that! I'm also going to add that as a benchmark

Comment (astrofrog) [reply astropy/units/tests/test_units.py]: Isn't it good to have a complex case though? It's only 3 seconds ;)

File: astropy/units/tests/test_units.py (hunk 1 above)
Comment (adrn) [reply astropy/units/tests/test_units.py]: I second this confusion!

Comment (mhvk) [reply astropy/units/tests/test_units.py]: this was added as part of #4849, which was to speed up the unit tests! Must have been some oversight.

File: astropy/units/tests/test_units.py (hunk 2 above)
Comment (mhvk) [reply astropy/units/tests/test_units.py]: Complex cases are tested elsewhere - but good idea to add it as a benchmark

**Pull request merged**